packages/terra-form-input/src/Input.jsx

```diff
@@ -120,13 +125,18 @@ class Input extends React.Component {
 
     const attributes = { ...customProps };
     const theme = this.context;
+
+    // unsupportedTypes: 'date', 'datetime-local', 'month', 'time', 'week', 'color', 'file', 'range', 'checkbox', 'radio', 'button', 'reset', 'submit'
+    const supportedTypes = [undefined, 'text', 'number', 'password', 'email', 'search', 'tel', 'url', 'hidden'];
+
+    const cxSupportedTypeClassNames = supportedTypes.includes(type) && cx(
```

Comment:
@benbcai something like this?
```suggestion
    const cxSupportedTypeClassNames = (supportedTypes.includes(type) || type === undefined ) && cx(
```

Reply:
@neilpfeiffer Yes!

Reply:
Curious - would this short circuit faster and skip needing to do the array.includes() if type is not set? 
Since `undefined` is defaulted and will probably be the most common case:

```suggestion
    const cxSupportedTypeClassNames = (type === undefined || supportedTypes.includes(type)) && cx(
```

Reply:
Would a defaultProp value to negate this case help? Or do we want to rely on the native type fallback?

Reply:
Updated here https://github.com/cerner/terra-core/pull/3386/commits/6ccb549abcd88c1ad6d0af5ea0318a0d38da4652